async-query-core/src/main/java/org/opensearch/sql/spark/utils/SQLQueryUtils.java

```diff
@@ -77,32 +83,73 @@ public static boolean isFlintExtensionQuery(String sqlQuery) {
     }
   }
 
-  public static List<String> validateSparkSqlQuery(String sqlQuery) {
-    SparkSqlValidatorVisitor sparkSqlValidatorVisitor = new SparkSqlValidatorVisitor();
-    SqlBaseParser sqlBaseParser =
-        new SqlBaseParser(
-            new CommonTokenStream(new SqlBaseLexer(new CaseInsensitiveCharStream(sqlQuery))));
-    sqlBaseParser.addErrorListener(new SyntaxAnalysisErrorListener());
+  public static List<String> validateSparkSqlQuery(DataSource datasource, String sqlQuery) {
     try {
-      SqlBaseParser.StatementContext statement = sqlBaseParser.statement();
-      sparkSqlValidatorVisitor.visit(statement);
-      return sparkSqlValidatorVisitor.getValidationErrors();
-    } catch (SyntaxCheckException syntaxCheckException) {
+      SqlBaseParser sqlBaseParser =
+          new SqlBaseParser(
+              new CommonTokenStream(new SqlBaseLexer(new CaseInsensitiveCharStream(sqlQuery))));
+      sqlBaseParser.addErrorListener(new SyntaxAnalysisErrorListener());
+      SqlBaseValidatorVisitor sqlParserBaseVisitor = getSparkSqlValidatorVisitor(datasource);
+      StatementContext statement = sqlBaseParser.statement();
+      sqlParserBaseVisitor.visit(statement);
+      return sqlParserBaseVisitor.getValidationErrors();
+    } catch (SyntaxCheckException e) {
+      logger.error(
+          String.format(
+              "Failed to parse sql statement context while validating sql query %s", sqlQuery),
+          e);
```

Comment:
Why do we add this error log?

Reply:
for better debuggability if there is a syntax error?

Reply:
As normal flow go through here, the log level should be `info` or `debug`.

Reply:
I am not sure if this is an info or debug level log
Imagine a bad sql query we are debugging in a log deep dive. It would be easier to trace it correctly if we throw error logs and user can also keep an eye on them.
this is an `error` level log because we have caught an exception from a user input or from another system. I strongly feel this should be retained as `error` level log

Reply:
I thought this skips SyntaxCheckException intentionally. @vamsi-amazon any thoughts?

Reply:
Yeah, I was little apprehensive to add error logs because..sometimes g4 files are not updated to the latest one from spark repo and this can result in syntax errors for valid queries. So, in case of syntax exceptions, I have decided to forward the query to spark in case of validation errors.

Now since we are clear on which spark version we are going to support, we should take g4 files from that branch and throw validation error in case of exception. what do you think @ykmr1224 ?